### PR TITLE
boost 1.87.0

### PR DESCRIPTION
**Pull request by cho-m**
Pre-release testing to check on linkage in preparation for only shipping multithreaded libraries (i.e. #182995).

Stable release planned for **Wednesday, December 11**.

Dependents bumped based on Linux bottle linkage (which always uses fully versioned libraries) and macOS `libboost*-mt.dylib` linkage.

---

Build failures (mostly due to Boost Asio removed APIs):

- [ ] `fastnetmon` - https://github.com/pavel-odintsov/fastnetmon/issues/1027
- [ ] `monero` - https://github.com/monero-project/monero/issues/9596
- [ ] `cpprestsdk` - https://github.com/microsoft/cpprestsdk/issues/1323 and https://github.com/microsoft/cpprestsdk/pull/1339
  - `azure-storage-cpp`
  - `vineyard`
- [ ] `osrm-backend` - https://github.com/Project-OSRM/osrm-backend/issues/7072 / https://github.com/Project-OSRM/osrm-backend/pull/7073
- [x] `uhd` - https://github.com/EttusResearch/uhd/issues/812
  - `gnuradio`
- [x] `i2pd` - https://github.com/PurpleI2P/i2pd/issues/2126
- [x] `dwarfs` - https://github.com/mhx/dwarfs/pull/245

### CODE REVIEW

File: Formula/b/boost-mpi.rb [hunk 1]
```diff
@@ -44,6 +44,9 @@ def install
     args << "cxxflags=-std=c++11"
     args << "cxxflags=-stdlib=libc++" << "linkflags=-stdlib=libc++" if ENV.compiler == :clang
 
+    # Avoid linkage to boost container and graph modules
+    args << "linkflags=-Wl,-dead_strip_dylibs" if OS.mac?
```
Comment (cho-m) [inline Formula/b/boost-mpi.rb:48]: Checking on this upstream - https://github.com/boostorg/boost/issues/985

**Comment by mbr0wn** [comment]
The next stable release of UHD will contain all the required updates. The changes will be pushed the UHD `master` branch soon. Thanks for tracking this!

**Review by carlocab — APPROVED** [review]
`tap_syntax` error needs fixing, but otherwise LGTM

File: Formula/m/mkvtoolnix.rb [hunk 2]
```diff
@@ -27,29 +28,40 @@ class Mkvtoolnix < Formula
...
   uses_from_macos "ruby" => :build
   uses_from_macos "zlib"
 
+  on_macos do
+    depends_on "gettext"
+  end
+
+  on_linux do
+    depends_on "gettext" => :build
+  end
```
Comment (carlocab) [inline Formula/m/mkvtoolnix.rb:57]: Seems like you could just make this unconditional given the `on_macos` block above.

Comment (cho-m) [reply Formula/m/mkvtoolnix.rb:57]: Yeah. Can update if another run is needed.

**Review by chenrui333 — APPROVED** [review]

**Comment by carlocab** [comment]
Somehow picked up a conflict with `libphonenumber`?

**Review by carlocab — APPROVED** [review]

**Comment by cho-m** [comment]
Another conflict. Going to go the old school way and manually block these #201069

**Comment by cho-m** [comment]
Going to try without self-hosted in case cache is still available. Will need at least a day for self-hosted runner.

**Review by carlocab — APPROVED** [review]

**Review by carlocab — APPROVED** [review]

**Pull request merged**